src/app/trade/_components/swap-action-button.tsx

```diff
@@ -43,7 +53,11 @@ export default function SwapActionButton(props: SwapActionButtonProps) {
     );
   }
 
-  return <Button className={commonSwapActionButtonClassName}>Swap</Button>;
+  return (
+    <Button className={commonSwapActionButtonClassName} onClick={() => swap()}>
```

Comment:
NIT : I think we can do that directly
```suggestion
    <Button className={commonSwapActionButtonClassName} onClick={swap}>
```

Reply:
I Can't unfortunately, Button tries to pass an event to the function